frontend/model/contracts/group.js

```diff
@@ -395,30 +422,18 @@ DefineContract({
         Vue.delete(state.profiles, data.member)
       },
       async sideEffect ({ data }) {
-        const rootState = sbp('state/vuex/state')
-        const contracts = rootState.contracts || {}
-
-        if (data.member === rootState.loggedIn.username) {
-          // If this member is re-joining the group, ignore the rest
-          // so the member doesn't remove themself again.
-          if (sbp('okTurtles.data/get', 'JOINING_GROUP')) {
-            return
-          }
-
-          const groupIdToSwitch = Object.keys(contracts)
-            .find(contractID => contracts[contractID].type === 'group' &&
-              contractID !== data.groupId &&
-              rootState[contractID].settings) || null
-
-          sbp('state/vuex/commit', 'setCurrentGroupId', groupIdToSwitch)
-          sbp('state/vuex/commit', 'removeContract', data.groupId)
-          sbp('controller/router').push({ path: groupIdToSwitch ? '/dashboard' : '/' })
-          // TODO - #828 remove other group members contracts if applicable
-        } else {
-          // TODO - #828 remove the member contract if applicable.
-          // sbp('state/vuex/commit', 'removeContract', data.memberID)
-        }
-        // TODO - #850 verify open proposals and see if they need some re-adjustment.
+        removeMemberSideEffect(data.member, data.groupId)
+      }
+    },
+    'gi.contracts/group/removeOurselves': {
+      validate: objectOf({
+        groupId: string
+      }),
+      process ({ data, meta }, { state }) {
+        Vue.delete(state.profiles, meta.username)
+      },
+      async sideEffect ({ data, meta }) {
+        removeMemberSideEffect(meta.username, data.groupId)
```

Comment:
I thought `await` was needed only if the function being called is async. In this case, `gi.sideEffects/group/removeMember` is not an async function (or at least ATM is not doing any async work). That's why I didn't add the `await`.

Anyway. I added it now, but would be nice to understand why it's really needed in this case.

ESlint has a rule to catch when we use `await` inside a function that isn’t `async`. I think a linter rule to detect if we call an async function without using `await` doesn’t exist.

Reply:
> Anyway. I added it now, but would be nice to understand why it's really needed in this case.

You are right that it isn't needed - I either didn't notice or forgot that `'removeMember'` did nothing asynchronous.

Side effects generally tend to do something asynchronous, since they are modifying state outside of contract, however this apparently isn't always the case, as shown here.

Perhaps it's still good convention though to get used to marking them all as async and using them as such, since this is how side effects are treated in general. To see what I mean by that, have a look in `state.js` to see how these side effects get called. You'll see this:

```js
      if (sbp('sbp/selectors/fn', `${selector}/sideEffect`)) {
        await sbp(`${selector}/sideEffect`, mutation)
      }
```

In other words, all side effects are awaited on. So this is why they must be defined as asynchronous, because it's assumed they are, and they're invoked as such.

Reply:
Hm.. okay, it can be for convenience/consistency, although it isn't needed